Controls/ConnectionControl.cs

```diff
@@ -130,21 +130,40 @@ private void CMB_sysid_SelectedIndexChanged(object sender, EventArgs e)
                     MainV2.comPort.sysidcurrent = temp.sysid;
                     MainV2.comPort.compidcurrent = temp.compid;
 
-                    if (MainV2.comPort.MAV.param.Count == 0 && !(Control.ModifierKeys == Keys.Control))
-                        MainV2.comPort.getParamList();
+                    //force param update and reload view to prevent missing information in config pages
+                    MainV2.comPort.getParamList();
+                    MainV2.View.Reload();
                 }
             }
         }
 
         private void cmb_sysid_Format(object sender, ListControlConvertEventArgs e)
         {
             var temp = (port_sysid) e.Value;
+            MAVLink.MAV_COMPONENT compid = (MAVLink.MAV_COMPONENT)temp.compid;
+            string mavComponentHeader = "MAV_COMP_ID_";
+            string mavComponentString = null;
 
             foreach (var port in MainV2.Comports)
             {
                 if (port == temp.port)
                 {
-                    e.Value = temp.port.BaseStream.PortName + "-" + port.MAVlist[temp.sysid, temp.compid].aptype.ToString() + "-" + ((int)temp.sysid);
+                    if (compid == (MAVLink.MAV_COMPONENT)1)
+                    {
+                        //use Autopilot type as displaystring instead of "FCS1"
+                        mavComponentString = port.MAVlist[temp.sysid, temp.compid].aptype.ToString();
+                    }
+                    else
+                    {
+                        //use name from enum if it exists, use the component ID otherwise
+                        mavComponentString = compid.ToString();
+                        if (mavComponentString.Length > mavComponentHeader.Length)
+                        {
+                            //remove "MAV_COMP_ID_" header
+                            mavComponentString = mavComponentString.Remove(0, mavComponentHeader.Length);
+                        }
+                    }
+                    e.Value = ((int)temp.sysid) + "-" + mavComponentString.Replace("_"," ");

```

Comment:
there is no distiction on which physical interface this is comming in on. ie some people use 2 links via diffrent methods to the same vechile.

Reply:
Is this currently possible? How would I go about testing this? Would adding the port back into the display string resolve this?

Reply:
@meee1 could you give us some feedback so that we can determine how best to proceed with the work we're currently doing?

Reply:
@JWangAeronavics, in lieu of feedback from @meee1, can we just update the string to contain something identifying the physical port?  "S1C1 (ARDUPILOT) on UDP:14550" and "S1C57 (UART_BRIDGE) on COM6" perhaps.